distributed/executor.py

```diff
@@ -1798,3 +1829,62 @@ def futures_of(o, executor=None):
             raise CancelledError(bad)
 
     return list(futures)
+
+
+@contextmanager
+def serialize_for_publish(x):
```

Comment:
This approach seems dangerous.  We might instead try to make a copy with `from copy import copy`